include/range/v3/utility/swap.hpp

```diff
@@ -42,10 +53,26 @@ namespace ranges
         /// \cond
         namespace adl_swap_detail
         {
-            using std::swap;
+
+            // The following implementation of swap is adapted from libc++:
+            template <class T>
+            RANGES_CXX14_CONSTEXPR
+            typename std::enable_if<
+                std::is_move_constructible<T>::value &&
+                std::is_move_assignable<T>::value
+             >::type
+            swap(T& x, T& y)
+            noexcept(std::is_nothrow_move_constructible<T>::value &&
+                     std::is_nothrow_move_assignable<T>::value)
+            {
+                T t(std::move(x));
+                x = std::move(y);
+                y = std::move(t);
+            }
```

Comment:
Done. 

I've had to remove the constexpr `iter_swap` and `swap` tests since `std::swap` will be picked up for, among others, fundamental types. The associated namespaces of fundamental types are empty, and there is not much that can be done about this. As a consequence, e.g. `iter_swap(int*, int*)` won't be constexpr. I don't know how large of a hit this will be, but I guess it will be a pretty large hit (see the commented out code in the `constexpr_core.cpp` tests). One can work around this for a user defined type (e.g. see `test/array.hpp`), but that workaround is just wrong (I just left it for exposition only).